Add missing lv_obj_class_t externs

amirgon opened the pull request:
The Micropython bindings assumes each LVGL object has a 'class' extern in the form of lv_objectname_class

Related: https://github.com/lvgl/lv_binding_micropython/issues/135

amirgon commented:
Actually I'm not sure about defining multiple classes to `lv_obj_class`.
That would cause ambiguity (more than one widget type has the same class) so we wouldn't be able to translate from class back to widget.  

Would it make sense to require **all** widgets to have proper class, including `lv_calendar_header_arrow`, `lv_calendar_header_dropdown` and `lv_spinner`?

embeddedt commented:
I think that is necessary for widget type-checking to work in C as well.

amirgon commented:
> Would it make sense to require **all** widgets to have proper class, including `lv_calendar_header_arrow`, `lv_calendar_header_dropdown` and `lv_spinner`?

I've added classes for all widget where class was missing.  
In cases where there is no constructor I've left it `NULL`.  
A distinct class for each widget is required for https://github.com/lvgl/lv_binding_micropython/issues/135

kisvegabor commented:
In 3fb8baf I implemented the missing constructors and added the missing `extern` you have shown in this PR.

amirgon commented:
> In [3fb8baf](https://github.com/lvgl/lvgl/commit/3fb8baf503411e006765020f60f295a4be16ba2d) I implemented the missing constructors and added the missing `extern`a you have shown in this PR.

Great!
Thanks.
